File: Haskell/Atividade V/problema11.hs
primeiros :: Int -> [t] -> [t]
primeiros _ [] = []
primeiros 0 _ = []
primeiros x (a:b) = (a : primeiros (x-1) b)

main = do
    print(primeiros 3 [1,2,3,4,5,6,7,8,9,10])

File: PROLOG/Atividade XVI/problema11.pl
concatena([],L,L).
concatena([H|T],L2,[H|LConcatenada]) :- concatena(T,L2,LConcatenada).

primeiros(_,[],[]).
primeiros(0,_,[]).
primeiros(N,[H|T],L) :-
    N =:= 0,!,
    primeiros(N-1,T,L1),
    concatena([H],L1,L),!.
% primeiros(3,[1,2,3,4,5],L).